### PR TITLE
JS-3: Implement home page

### PR DESCRIPTION
**Pull request by TeodorTunev**

### CODE REVIEW

File: app/components/slider/slider.component.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<div class="slider" (window:resize)="resize()">
+  <div class="slider-container" (swipeleft)="moveItemContainer(true)" (swiperight)="moveItemContainer(false)">
+      <div *ngFor="let job of jobs" class="item-container">
+        <img  class="company-logo no-select" [src]="(job.company.companyLogo && job.company.companyLogo.imageUrlSmall) || 'app/assets/images/placeholder-logo.png'">
```
Comment (belev) [inline app/components/slider/slider.component.html:4]: remove second whitespace after img tag

File: app/components/slider/slider.component.html [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<div class="slider" (window:resize)="resize()">
```
Comment (belev) [inline app/components/slider/slider.component.html]: rename resize handler to onResize

File: app/components/slider/slider.component.html [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  <div class="slider-container" (swipeleft)="moveItemContainer(true)" (swiperight)="moveItemContainer(false)">
```
Comment (belev) [inline app/components/slider/slider.component.html]: moveItemContainer -> onSwipe

File: app/components/slider/slider.component.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
+import {Component, OnInit, ElementRef, Input} from '@angular/core';
```
Comment (belev) [inline app/components/slider/slider.component.ts]: use double quotes

File: app/components/slider/slider.component.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    templateUrl: 'slider.component.html',
+    styleUrls: ['slider.component.css']
+})
+export class SliderComponent implements OnInit {
+  @Input() jobs: Array<Job>;
+  private slider: any;
+  private sliderContainer: any;
+  private firstItemContainer: any;
+  private lastFullyVisualizatedItemIndex: number;
+  isCompanyUser: boolean = false;
```
Comment (belev) [inline app/components/slider/slider.component.ts:16]: get value from authManager function

File: app/components/slider/slider.component.ts [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+  constructor(private sliderElement: ElementRef) {
+  }
+
+  ngOnInit() {
+    this.slider = this.sliderElement.nativeElement.getElementsByClassName('slider')[0];
+    this.sliderContainer = this.sliderElement.nativeElement.getElementsByClassName('slider-container')[0];
+  }
+
+  moveItemContainer(isMoveToLeft: boolean) {
```
Comment (belev) [inline app/components/slider/slider.component.ts]: isMoveToLeft -> isLeftSwipe

File: app/components/slider/slider.component.ts [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    let firstItemContainerOffsetWidth = this.getFirstItemContainer().offsetWidth;
+    let maxVisualizatedItems = this.getMaxVisualisationItemContainers(firstItemContainerOffsetWidth);
+    if (!this.lastFullyVisualizatedItemIndex) {
+      this.lastFullyVisualizatedItemIndex = maxVisualizatedItems;
+    }
+
+    let sliderContainerLeftPosition = parseInt(this.sliderContainer.style.left || 0);
+    let pixelsValueForMovingSliderContainer = parseInt(firstItemContainerOffsetWidth * maxVisualizatedItems, 10);
+
+    if (this.lastFullyVisualizatedItemIndex + maxVisualizatedItems >= this.jobs.length && isMoveToLeft) {
```
Comment (belev) [inline app/components/slider/slider.component.ts]: add brackets around: this.lastFullyVisualizatedItemIndex + maxVisualizatedItems >= this.jobs.length

File: app/components/slider/slider.component.ts [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+      this.sliderContainer.style.left = (((this.jobs.length * firstItemContainerOffsetWidth) - this.slider.offsetWidth) * -1) + 'px';
+      this.lastFullyVisualizatedItemIndex = this.jobs.length;
+    } else if (this.lastFullyVisualizatedItemIndex - maxVisualizatedItems <= maxVisualizatedItems && !isMoveToLeft) {
```
Comment (belev) [inline app/components/slider/slider.component.ts]: add brackets around: this.lastFullyVisualizatedItemIndex - maxVisualizatedItems <= maxVisualizatedItems

File: app/components/slider/slider.component.ts [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+      let indexMultiplier = isMoveToLeft ? 1 : -1;
+      this.sliderContainer.style.left = (sliderContainerLeftPosition + ((pixelsValueForMovingSliderContainer - 10) * directionMultiplier)) + 'px';
+      this.lastFullyVisualizatedItemIndex  = this.lastFullyVisualizatedItemIndex + (maxVisualizatedItems * indexMultiplier);
+    }
+
+  }
+
+
+  resize(event) {
+    //TODO: make responsivnes on resizing
```
Comment (belev) [inline app/components/slider/slider.component.ts]: responsivnes -> responsiveness

File: app/components/slider/slider.component.ts [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  private getFirstItemContainer(): any {
+    return this.firstItemContainer = this.sliderElement.nativeElement.getElementsByClassName('item-container')[0];
```
Comment (belev) [inline app/components/slider/slider.component.ts]: remove: this.firstItemContainer =

File: app/models/job/job.ts [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+}
+
+class HourlyPay {
+  active: boolean;
+  currency: string;
+  grossSalary: number;
+  id: string;
+  netSalary: number;
+  rateExcludingVAT: number;
+  rateIncludingVAT: number;
```
Comment (belev) [inline app/models/job/job.ts:55]: add empty line

File: app/services/job-proxy.service.ts [hunk 12]
```diff
@@ -0,0 +1,15 @@
+import { Injectable } from '@angular/core';
```
Comment (belev) [inline app/services/job-proxy.service.ts]: remove spaces and use double quotes

File: app/styles/main.scss [hunk 13]
```diff
@@ -205,4 +206,14 @@ body {
       font-size: 12px;
     }
   }
+
```
Comment (belev) [inline app/styles/main.scss:209]: remove one empty line

File: app/views/home/home.component.html [hunk 14]
```diff
@@ -1,4 +1,92 @@
...
+  </section>
+  <section class="job-section">
+    <div class="job-section-header">
+      <div class="calendar-container">
+        <div class="calendar">
+          {{today}}
+        </div>
+        {{'start.job_section.title' | translate}}
+      </div>
+      <div class="all-jobs-button-container"><a href="#"><span>{{'common.show_all' |translate}}</span></a></div>
```
Comment (belev) [inline app/views/home/home.component.html]: add whitespace: |translate

File: app/views/home/home.component.html [hunk 15]
```diff
@@ -1,4 +1,92 @@
...
+    </div>
+    <slider [jobs]="newJobs"></slider>
+
+  </section>
+  <div class="instructions-container">
+    <section class="instruction-pane">
+      <h3 class="heading">{{'start.steps_section.title' | translate}}</h3>
+      <!--<p>Description</p>-->
```
Comment (belev) [inline app/views/home/home.component.html]: delete

File: app/views/home/home.component.html [hunk 16]
```diff
@@ -1,4 +1,92 @@
...
+      <ul class="list">
+        <li class="item">{{'start.instruction_section.slide_2.text_2' | translate}}</li>
+        <li class="item">{{'start.instruction_section.slide_2.text_3' | translate}}</li>
+        <li class="item">{{'start.instruction_section.slide_2.text_4' | translate}}</li>
+      </ul>
+      <a href="#" class="button">{{'common.start_here' | translate}}</a>
+    </section>
+  </div>
+  <section class="section-partners">
+    <div class="partners-title"> {{'start.partner_section.title' | translate}} </div>
```
Comment (belev) [inline app/views/home/home.component.html]: remove empty space after: translate}}

File: app/views/home/home.component.html [hunk 17]
```diff
@@ -1,4 +1,92 @@
...
+    <p class="partners-text"> {{'start.partner_section.text' | translate}} </p>
```
Comment (belev) [inline app/views/home/home.component.html]: remove empty space after: translate}}

File: app/views/home/home.component.scss [hunk 18]
```diff
@@ -0,0 +1,274 @@
...
+          margin-left: 6px;
+        }
+
+        a {
+          text-decoration: none;
+        }
+      }
+    }
+  }
+
```
Comment (belev) [inline app/views/home/home.component.scss:188]: remove empty lines

File: app/views/home/home.component.scss [hunk 19]
```diff
@@ -0,0 +1,274 @@
...
+  .section-partners::before {
+    content: "";
+    display: block;
+    height: 100%;
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100%;
+
```
Comment (belev) [inline app/views/home/home.component.scss:243]: remove empty line

File: app/views/home/home.component.ts [hunk 20]
```diff
@@ -1,23 +1,36 @@
...
   password: string;
-  constructor(private authManager: AuthManager) {
+  newJobs: Array<Job>;
+  isCompanyUser: boolean;
+  constructor(private jobProxy: JobProxy, private authManager: AuthManager) {
+    this.isCompanyUser = authManager.isCompanyUser();
   }
 
   ngOnInit() {
+    this.jobProxy.getJobs({include: 'company,hourly-pay,company.company-images', 'filter[filled]':false}).then(result => {
```
Comment (belev) [inline app/views/home/home.component.ts:24]: add whitespace after: 'filter[filled]':

File: app/views/home/home.component.ts [hunk 21]
```diff
@@ -1,23 +1,36 @@
...
+      this.newJobs = result;
+      console.log(result);
```
Comment (belev) [inline app/views/home/home.component.ts]: delete

File: app/views/home/home.component.ts [hunk 22]
```diff
@@ -1,23 +1,36 @@
...
 
 @Component({
   moduleId: module.id,
-  templateUrl: 'home.component.html'
+  templateUrl: 'home.component.html',
+  styleUrls: ['home.component.css'],
+  providers: [JobProxy]
 })
 export class HomeComponent implements OnInit {
+  today: number;
```
Comment (belev) [inline app/views/home/home.component.ts]: add initialization here: = new Date().getDate();

File: app/views/home/home.component.ts [hunk 23]
```diff
@@ -1,23 +1,36 @@
...
+  constructor(private jobProxy: JobProxy, private authManager: AuthManager) {
+    this.isCompanyUser = authManager.isCompanyUser();
   }
 
   ngOnInit() {
+    this.jobProxy.getJobs({include: 'company,hourly-pay,company.company-images', 'filter[filled]':false}).then(result => {
+      this.newJobs = result;
+      console.log(result);
+    });
+    this.today = new Date().getDate();
```
Comment (belev) [inline app/views/home/home.component.ts]: delete

File: app/components/slider/slider.component.ts [hunk 24]
```diff
@@ -0,0 +1,64 @@
...
+      this.lastFullyVisualizatedItemIndex = maxVisualizatedItems;
+    }
+
+    let sliderContainerLeftPosition = parseInt(this.sliderContainer.style.left || 0);
+    let pixelsValueForMovingSliderContainer = parseInt(firstItemContainerOffsetWidth * maxVisualizatedItems);
+
+    if ((this.lastFullyVisualizatedItemIndex + maxVisualizatedItems >= this.jobs.length) && isLeftSwipe) {
+      this.sliderContainer.style.left = (((this.jobs.length * firstItemContainerOffsetWidth) - this.slider.offsetWidth) * -1) + 'px';
+      this.lastFullyVisualizatedItemIndex = this.jobs.length;
+    } else if (this.lastFullyVisualizatedItemIndex - maxVisualizatedItems <= maxVisualizatedItems && !isLeftSwipe) {
```
Comment (belev) [inline app/components/slider/slider.component.ts:40]: add brackets around: this.lastFullyVisualizatedItemIndex - maxVisualizatedItems <= maxVisualizatedItems

File: app/components/slider/slider.component.ts [hunk 25]
```diff
@@ -0,0 +1,64 @@
...
+      this.lastFullyVisualizatedItemIndex = maxVisualizatedItems;
+    } else {
+      let directionMultiplier = isLeftSwipe ? -1 : 1;
+      let indexMultiplier = isLeftSwipe ? 1 : -1;
+      this.sliderContainer.style.left = (sliderContainerLeftPosition + ((pixelsValueForMovingSliderContainer - 10) * directionMultiplier)) + 'px';
+      this.lastFullyVisualizatedItemIndex  = this.lastFullyVisualizatedItemIndex + (maxVisualizatedItems * indexMultiplier);
+    }
+
+  }
+
```
Comment (belev) [inline app/components/slider/slider.component.ts:51]: remove empty line

**Review by belev — APPROVED** [review]

File: index.html [hunk 26]
```diff
@@ -19,9 +19,11 @@
   <script src="node_modules/zone.js/dist/zone.js"></script>
   <script src="node_modules/reflect-metadata/Reflect.js"></script>
   <script src="node_modules/systemjs/dist/system.src.js"></script>
+
   <!-- endbuild -->
 
   <!-- build:app -->
+  <script src="node_modules/hammerjs/hammer.min.js"></script>
```
Comment (buren) [inline index.html:26]: Love the dedication for 🚀 :)

**Pull request merged**